File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def recursion(numbers, target, idx, curr):
    global answer
        
    if idx == len(numbers):
        if curr == target:
            answer += 1
        return
    
    # + 연산
    recursion(numbers, target, idx+1, curr + numbers[idx])
    
    # - 연산
    recursion(numbers, target, idx+1, curr - numbers[idx])


def solution(numbers, target):
    global answer
    answer = 0
    
    recursion(numbers, target, 0, 0)
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
let answer = 0

function recursion(target, numbers, idx, result) {
    if (idx === numbers.length) {
        if (result === target) {
            answer = answer + 1
        }
        return
    }
    
    // 1. 현재 수를 더하기
    recursion(target, numbers, idx + 1, result + numbers[idx])
    
    // 2. 현재 수를 빼기
    recursion(target, numbers, idx + 1, result - numbers[idx])
}

function solution(numbers, target) {
    recursion(target, numbers, 0, 0)
    
    return answer;
}